src/cliffordsynthesis/CliffordSynthesizer.cpp

```diff
@@ -523,8 +543,11 @@ std::shared_ptr<qc::QuantumComputation>
 CliffordSynthesizer::synthesizeSubcircuit(
     const std::shared_ptr<qc::QuantumComputation>& qc, std::size_t begin,
     std::size_t end, const Configuration& config) {
-  const Tableau       subTargetTableau{*qc, begin, end, true};
+  std::cout << "Synthesizing Subcircuit" << std::endl;
```

Comment:
This is the first line in the function that is called in the future that causes the SEGFAULT. However, this line is not even printed to the output. I am not sure, whether this is because no output escapes the future or because the error happens between calling the `get()` of the future and the invocation of this function inside the future. When I am testing locally, those lines in the future are visible as well.